[New Resource]: Broadcast Domain

acch opened the pull request:
Adds ability to create broadcast domains.

Closes #303.

Acceptance tests pass:
```shell
$ TF_ACC=1 go test ./internal/provider/networking/network_broadcast_domain_resource_test.go -v
=== RUN   TestAccNetworkBroadcastDomainResource
--- PASS: TestAccNetworkBroadcastDomainResource (3.73s)
PASS
ok      command-line-arguments  3.733s
```

Example Terraform Configuration:
```terraform
// sample broadcast domain with required properties
resource "netapp-ontap_network_broadcast_domain" "example" {
  cx_profile_name = ...
  ipspace         = "Default"
  name            = "bd1"
  mtu             = "1500"
}
```

acch committed:
feat: add `broadcast_domain` data source

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add `broadcast_domains` data source

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add `broadcast_domain` resource

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
chore: minor formatting for consistency

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add `broadcast_domain` resource import functionality

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
chore: minor formatting for consistency

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add acceptance tests for `broadcast_domain` resource

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
chore: change cx_profile_name for consistency with other examples

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: use Terraform Set Type instead of Go built-in slice

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: consistent error handling

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
feat: add remaining attribute description

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
chore: finalize data model

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch committed:
chore: mark cx_profile_name as unable to update in-place

Signed-off-by: Achim Christ <achim.christ@sva.de>

carchi8py reviewed (changes requested):
Looks good. Only thing missing is the documentation pages.

acch committed:
feat: add `broadcast_domain` documentation

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch commented:
Thanks @carchi8py for your feedback!

I've manually added the documentation pages in the `docs/` directory. This is very tedious; isn't it possible to generate the docs automatically? The TF scaffolding template contains scripts for that ([here](https://github.com/hashicorp/terraform-provider-scaffolding-framework/blob/main/tools/tools.go)), but I see that you've removed them...

acch commented:
@carchi8py Are we good to merge this PR?

Please let me know what I need to do exactly to have the PRs added to the provider. My understanding is that I've addressed all requested changes... please let me know what's missing.

Many thanks!

suhasbshekar commented:
Can you update CHANGELOG.md file?

Also your branch is behind the integration/main, we added few more checks to the pipeline as it was missing here, since your PR is from a fork of this repo, so could you pull the new changes into your branch and update the PR?

Just for our reference, since we have disabled ACC tests as our testing profiles are down, acc tests are passing for this PR when tested locally.

acch committed:
chore: update changelog

Signed-off-by: Achim Christ <achim.christ@sva.de>

acch commented:
@suhasbshekar Thanks for your feedback.

- I've added an entry to CHANGELOG.md
- I've rebased my branch to integration/main (once again)

Acc tests are passing on my local test system (ONTAP Select running 9.15.1P4):

```shell
$ date && TF_ACC=1 go test ./internal/provider/networking/network_broadcast_domain_resource_test.go -v && date
Tue Feb  4 09:26:22 UTC 2025
=== RUN   TestAccNetworkBroadcastDomainResource
--- PASS: TestAccNetworkBroadcastDomainResource (3.65s)
PASS
ok      command-line-arguments  3.657s
Tue Feb  4 09:26:26 UTC 2025
```

Please let me know what else is needed.
Many thanks!

suhasbshekar commented:
Please resolve the conflicts.

acch commented:
@suhasbshekar I've rebased the PR once again.

Can you please let me know when you'll be able to merge these changes?

suhasbshekar commented:
Conflicts for this PR
